modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/AbstractExtensionObjectFactory.java

```diff
@@ -223,6 +230,10 @@ public void resolveParameters(ObjectType objectType, DefaultObjectBuilder builde
 
       if (valueResolver != null) {
         builder.addPropertyResolver(objectField.getName(), valueResolver);
+      } else if (field.isRequired() && !field.getAnnotation(FlattenedTypeAnnotation.class).isPresent()) {
+        throw new IllegalStateException(format("Class %s contains field '%s' of type %s which is required but wasn't set",
```

Comment:
Error message shouldn't contain the terms class or field. From the user's point of view, he's missing a parameter. He doesn't know about the internal structure